src/lando/api/legacy/api/landing_jobs.py

```diff
@@ -27,31 +39,46 @@ def put(request: HttpRequest, landing_job_id: str, data: dict):
             updated (for example, when trying to cancel a job that is already in
             progress).
     """
-    with LandingJob.lock_table:
-        landing_job = LandingJob.objects.get(pk=landing_job_id)
+    data = json.loads(request.body)
+    data["landing_job_id"] = landing_job_id
+    form = LandingJobForm(data)
 
-    if not landing_job:
-        raise Http404(f"A landing job with ID {landing_job_id} was not found.")
+    if not form.is_valid():
+        data = {
+            "errors": [
+                f"{field}: {', '.join(field_errors)}"
+                for field, field_errors in form.errors.items()
+            ]
+        }
+        return JsonResponse(data, status=400)
+
+    landing_job_id = form.cleaned_data["landing_job_id"]
+    status = form.cleaned_data["status"]
+
+    with LandingJob.lock_table:
+        try:
+            landing_job = LandingJob.objects.get(pk=landing_job_id)
+        except LandingJob.DoesNotExist:
+            raise Http404(f"A landing job with ID {landing_job_id} was not found.")
```

Comment:
As discussed, we can either change this to a JsonResponse with a custom status code to make tests pass, or the tests can ignore the content of the response and just check the status code as a temporary solution. Though this can all be fixed as part of [bug 1929466](https://bugzilla.mozilla.org/show_bug.cgi?id=1929466).

Reply:
Yup, let's postpone to the other bug https://bugzilla.mozilla.org/show_bug.cgi?id=1929466#c2